backend/src/v2/compiler/argocompiler/common.go

```diff
@@ -46,6 +56,80 @@ var commonEnvs = []k8score.EnvVar{{
 	},
 }}
 
+func GetMLPipelineServiceTLSEnabled() (bool, error) {
+	mlPipelineServiceTLSEnabledStr := os.Getenv(MLPipelineTLSEnabledEnvVar)
+	if mlPipelineServiceTLSEnabledStr == "" {
+		return DefaultMLPipelineTLSEnabled, nil
+	}
+	mlPipelineServiceTLSEnabled, err := strconv.ParseBool(os.Getenv(MLPipelineTLSEnabledEnvVar))
+	if err != nil {
+		return false, err
+	}
+	return mlPipelineServiceTLSEnabled, nil
+}
+
+// ConfigureCABundle adds CABundle environment variables and volume mounts
+// if CA Bundle env vars are specified.
+func ConfigureCABundle(tmpl *wfapi.Template) {
+	caBundleCertPath := os.Getenv(common.CaBundlePath)
+	caBundleDir := path.Dir(caBundleCertPath)
```

Comment:
I don't think we need to make this as configurable. We essentially just need the secret name and the key in the `data` field on the secret (default it to whatever cert-manager uses). The path can be hardcoded with a constant variable used here, driver, and launcher.